rust/azure_iot_operations_mqtt/src/session/session.rs

```diff
@@ -160,501 +182,339 @@ impl Session {
     /// # Errors
     /// Returns a [`SessionError`] if the session encounters a fatal error and ends.
     pub async fn run(mut self) -> Result<(), SessionError> {
-        self.state.transition_running();
```

Comment:
do we still want to call this state transition?

Reply:
Correct, this is one of the parts that will get removed